hedera-node/hedera-util-service-impl/src/main/java/com/hedera/node/app/service/util/impl/handlers/UtilPrngHandler.java

```diff
@@ -18,27 +18,139 @@
 
 import static java.util.Objects.requireNonNull;
 
+import com.google.common.math.LongMath;
 import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.base.ResponseCodeEnum;
+import com.hedera.hapi.node.util.UtilPrngTransactionBody;
+import com.hedera.node.app.service.network.ReadableRunningHashLeafStore;
+import com.hedera.node.app.service.util.impl.config.PrngConfig;
+import com.hedera.node.app.service.util.impl.records.UtilPrngRecordBuilder;
+import com.hedera.node.app.service.util.records.PrngRecordBuilder;
+import com.hedera.node.app.spi.meta.HandleContext;
+import com.hedera.node.app.spi.workflows.PreCheckException;
 import com.hedera.node.app.spi.workflows.PreHandleContext;
 import com.hedera.node.app.spi.workflows.TransactionHandler;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.crypto.RunningHash;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.concurrent.ExecutionException;
 import javax.inject.Inject;
 import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 /**
  * This class contains all workflow-related functionality regarding {@link
  * HederaFunctionality#UTIL_PRNG}.
  */
 @Singleton
 public class UtilPrngHandler implements TransactionHandler {
+    private static final Logger log = LogManager.getLogger(UtilPrngHandler.class);
+    public static final byte[] MISSING_BYTES = new byte[0];
+
     @Inject
     public UtilPrngHandler() {
         // Dagger2
     }
 
-    @Override
```

Comment:
Please keep the `Override` annotation. It ensures that compiler and developer are immediately in the know.

(An `{@inheritDoc}` by itself on the other hand is redundant IMO, in particular javadoc does not need it. But I noticed some people insist on having it. Guess, it does not hurt either.)

Reply:
Yes, removed by mistake. Put it back